Cabal/src/Distribution/Simple/Configure.hs

```diff
@@ -1961,11 +1962,19 @@ checkPackageProblems :: Verbosity
 checkPackageProblems verbosity dir gpkg pkg = do
   ioChecks      <- checkPackageFiles verbosity pkg dir
   let pureChecks = checkPackage gpkg (Just pkg)
-      errors   = [ e | PackageBuildImpossible e <- pureChecks ++ ioChecks ]
-      warnings = [ w | PackageBuildWarning    w <- pureChecks ++ ioChecks ]
+      (errors, warnings) =
+        partitionEithers (M.mapMaybe classEW $ pureChecks ++ ioChecks)
   if null errors
-    then traverse_ (warn verbosity) warnings
-    else die' verbosity (intercalate "\n\n" errors)
+    then traverse_ (warn verbosity) (map show warnings)
```

Comment:
See below.  `Show` is for Haskell concrete syntax of data, for displaying to user use `pretty`.

Reply:
Ah, reading better, I undestand now: «do not introduce `show` in *internal* code». That is very reasonable. I can make a function for principled pretty printing and use it internally, if we decide to deprecate that `Show` instance later we can do it without rewriting some of the implementation.

Will make a PR this evening or tomorrow.